### ISSUE TITLE
Regarding the question of changing the output link score to link probability.

### ISSUE THREAD

**Issue by ShenchenYu98** [facebookresearch/SEAL_OGB#30]
Hi, I read your paper Link Prediction Based on Graph Neural Networks.This is a great job.
Currently I'm doing my own dataset experiments on your code.The output I want is the probability that the current link exists.I pass the last dimensional output of the DGCNN model through the sigmoid function as shown below.
#MLP.
#x = F.relu(self.lin1(x))
#x = F.dropout(x, p=0.5, training=self.training)
#x = self.lin2(x)
#x = torch.sigmoid(x)
I got this result.
![image](https://user-images.githubusercontent.com/89429652/179531704-62ff8efa-7d76-4297-97f4-5c86ae5485e0.png)
In additional experiments in your paper, you also performed mean precision comparisons.
I'm not sure if you set the classification threshold to 0.5 for this part of the experiment? That is, the value after passing the sigmoid function, the value greater than 0.5 is a true edge, and the value less than 0.5 is a false edge. Or do you have other tricks?
I saw a similar answer in a previous issue, but I'm more interested to know how the precision experiments in your paper are done.
I would appreciate it if you could answer my question!

**Comment by muhanzhang** [comment]
Hi! The average precision metric does not require a fixed threshold. In fact, it averages over the precision across all thresholds. See https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html

**Comment by ShenchenYu98** [comment]
> Hi! The average precision metric does not require a fixed threshold. In fact, it averages over the precision across all thresholds. See https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html

First of all, thank you very much for taking the time to answer my question.
I said the average precision before, maybe I misunderstood.
Actually, what I need is to be able to know exactly whether the input edge exists, so I may need to output the probability that this edge exists. This problem can be regarded as a binary classification problem in essence. When the last layer of the network outputs one-dimensional data, it gets a value between 0 and 1 through the sigmoid function. The result of my trained model is shown in the figure below.
![1658207540597](https://user-images.githubusercontent.com/89429652/179670146-bcf9a2d8-ad2f-4a74-85c7-6c123838aeb8.png)
If I want to calculate the accuracy evaluation, for the above output results, set the threshold to 0.5, so that the probability greater than 0.5 is judged as a true edge, is this reasonable?
I have seen many papers related to link prediction that use AUC as an evaluation metric in the final experiments, but rarely use ACC. why is it like this? I don't know if it is convenient for you to answer my question?
I'm a beginner in link prediction and some of the questions may sound naive to you, so please bear with me if I offend you.

**Comment by muhanzhang** [comment]
The ACC metric is sensitive to the specific threshold you set. Setting it to 0.5 is usually a reasonable choice. But a more valid solution should be to treat it as a hyperparameter and tune it using the validation set. 

The reason why most link prediction papers do not adopt ACC is because most real world networks are sparse, where negative links are way more than positive links in quantity. ACC is not robust to class imbalance, e.g., naively predicting all node pairs as negative will generate a very high ACC, but it makes no sense. Thus, AUC (which is robust to class imbalance) is adopted.

**Comment by ShenchenYu98** [comment]
> The ACC metric is sensitive to the specific threshold you set. Setting it to 0.5 is usually a reasonable choice. But a more valid solution should be to treat it as a hyperparameter and tune it using the validation set.
> 
> The reason why most link prediction papers do not adopt ACC is because most real world networks are sparse, where negative links are way more than positive links in quantity. ACC is not robust to class imbalance, e.g., naively predicting all node pairs as negative will generate a very high ACC, but it makes no sense. Thus, AUC (which is robust to class imbalance) is adopted.

Thank you so much for your answer, I learned a lot of new things. I have no problem. I will close this question after replying to this message.
